### DISCUSSION TITLE
How to set table grid height in version 1.0.9

### DISCUSSION THREAD

**Discussion by JS-Aibel** [zauberzeug/nicegui#224] [Q&A] [answered]
In the example https://nicegui.io/reference#table the grid height is set with the ui.table property `).classes('max-h-40').` When setting `classes('max-h-50')` the height of the grid increases. When setting the height to greater than 50, the height of the table does not change. 
Changing the height of the table to `.classes('max-h-100')` seems to work in version 0.0.9, but not in version 1.0.9.

**Comment by falkoschindler** [comment, marked as answer]
The class "max-h-40" is a [Tailwind](https://tailwindcss.com/) class. It is a convenient way to configure the style and layout of a web application without setting individual CSS properties.

But there are no Tailwind classes for *all* possible CSS values. E.g. there are [max-height](https://tailwindcss.com/docs/max-height) classes `max-h-40`, `max-h-44`, `max-h-48`, but not `max-h-50` or `max-h-100`.

This can change with newer versions of Tailwind. Since we updated from Tailwind 2 to Tailwind 3 with NiceGUI 1.0, the behavior might have changed.

In order to set some different max height, you can either set the element's style, e.g. `.style('max-height: 100px')`, or set a custom value [via Tailwind](https://tailwindcss.com/docs/adding-custom-styles#using-arbitrary-values), e.g. `.classes('max-w-[100px]')`.

**Comment by falkoschindler** [comment]
I just noticed another subtlety:

By default a `ui.table` has the classes "ag-theme-balham", "w-full" and "h-64". This is to obtain a reasonable default size, even if the user does not specify any class himself. But if you want to set the height yourself, you might need to remove the "h-64" class: `.classes('h-96', remove='h-64')`.

Using `.style()` circumvents this issue, since style definitions overrule classes.

**Reply by abulka** [reply]
> By default a `ui.table` has the classes "ag-theme-balham", ...
> 
> Using `.style()` circumvents this issue, since style definitions overrule classes.

Relatedly,  I was trying to apply a dark style to my `ui.aggrid` and had to
```python
.classes(add='ag-theme-alpine-dark', remove='ag-theme-balham')
```
just in case it helps anyone.

**Comment by cryptedx** [comment]
Unfortunatly this is not working for me. I removed `h-64` an tried to set `h-2/3` and a lot of other values but most of them are not working. The table completely disapears. Even when I set `.style("max-height: 500px")` then `h-64 is not being overriden` or removed.

![image](https://user-images.githubusercontent.com/20707078/213004994-e694cf4c-a3b3-4317-8eb4-bdd5f0fcd4d6.png)

**Reply by falkoschindler** [reply]
`.style("max-height: 500px")` will only define a [_maximum_ height](https://www.w3schools.com/cssref/pr_dim_max-height.php), not the height itself. So try setting `.style("height: 500px")`.

Note that using `.style("height: 500px")` won't remove classes like `h-64`. But the style definition will have precedence when rendering in the browser. So `h-64` is still in the class list, but has no effect.

This all might be a bit confusing and especially the default table height set to `h-64` might not be the most intuitive design decision... But I hope you can get it to work. If not, you can share the code which is not working as expected, which makes it easier to give specific advice.

**Comment by cryptedx** [comment]
Wow you are really fast! :-)

It's quite simple what I am trying to archive. I just want to grow the tables height.

```py
table = ui.table(
    {
        "defaultColDef": {
            "minWidth": 100,
            "resizable": True,
        },
        "columnDefs": [
            {...}
         ],
        "rowData": data_list,
    }
).classes("h-2/3", remove="h-64")
```
![image](https://user-images.githubusercontent.com/20707078/213009053-9c0eb01a-ff9d-45f3-891b-c8f0df4c38e5.png)

Unfortunately the table completely disappears. There is only a line visible:
![image](https://user-images.githubusercontent.com/20707078/213009146-39d8b042-cf02-4053-8898-d292f6af57f6.png)

When I use `h-screen` then table height changes, but then it's too high. Beside of that class I haven't found any other which is applying a height. All give me no table and a line like in the screenshot above.

Thanks!

**Reply by falkoschindler** [reply]
I see. The problem is: `h-2/3` translates to `height: 66.666667%;` which refers to the height of the surrounding container. But this is the page itself and it has a default height of 0. So everything collapses.

The following example defines a card with given height, yielding a page with 67% of its height:

```python
with ui.card().classes('w-full h-screen'):
    ui.table({}).classes('h-2/3', remove='h-64')
```

But it's probably easier to use a different height definition, e.g. `h-96` (the selection of pre-defined heights is rather limited), `h-[400px]` (yes, you can use custom classes with arbitrary height values) or even relative to the view port `h-[66.7vh]`. Or you use CSS with `height: 400px` or `height: 66.7vh`, respectively:

```python
ui.table({}).style('height: 66.67vh')
```

**Comment by cryptedx** [comment]
Thank you very much for your tips and explanations! With the latter code it worked immediately
